src/ast/parse.rs

```diff
@@ -4,16 +4,14 @@ use crate::*;
 use ordered_float::OrderedFloat;
 
 pub fn parse_program(filename: Option<String>, input: &str) -> Result<Vec<Command>, ParseError> {
-    let (out, _span, rest) = program(&Context::new(filename, input))?;
-    assert!(rest.is_at_end(), "did not parse entire program");
-    Ok(out)
+    let sexps = all_sexps(Context::new(filename, input))?;
```

Comment:
Do you mind moving the sexp parser into it's own file?

Reply:
Hmm, a little bit. Where should `Span` live? It's under 1000 lines and we have way worse offenders, whereas everything here (hopefully) is only in service of implementing `parse_program` and `parse_expr`.

Reply:
Yeah idk where Span should live
ok